interfaces/sourcegen/sourcegen/_HeaderFileParser.py

```diff
@@ -1,33 +1,90 @@
-"""Parser for existing CLib headers."""
+"""Parser for YAML header configurations or existing CLib headers."""
 
 # This file is part of Cantera. See License.txt in the top-level directory or
 # at https://cantera.org/license.txt for license and copyright information.
 
 from pathlib import Path
 import logging
 import re
-from typing import List
 
-from ._dataclasses import HeaderFile, Func
+from ._dataclasses import HeaderFile, Func, Recipe
+from ._helpers import read_config
 
 
-_logger = logging.getLogger()
+_LOGGER = logging.getLogger()
 
-_clib_path = Path(__file__).parent.joinpath("../../../include/cantera/clib").resolve()
-_clib_ignore = ["clib_defs.h", "ctmatlab.h"]
+_CLIB_PATH = Path(__file__).parents[3] / "include" / "cantera" / "clib"
+_CLIB_IGNORE = ["clib_defs.h", "ctmatlab.h"]
+
+_DATA_PATH = Path(__file__).parent / "_data"
 
 class HeaderFileParser:
+    """
+    Parser for header files or corresponding YAML specifications.
+
+    Provides for convenience methods to generate lists of `HeaderFile` objects, which
+    themselves are used for subsequent code scaffolding.
+    """
 
-    def __init__(self, path: Path, ignore_funcs: List[str] = None):
+    def __init__(self, path: Path, ignore_funcs: list[str] = None):
         self._path = path
         self._ignore_funcs = ignore_funcs
 
     @classmethod
-    def from_headers(cls, ignore_files, ignore_funcs) -> List[HeaderFile]:
+    def headers_from_yaml(cls, ignore_files, ignore_funcs) -> list[HeaderFile]:
+        """Parse header file YAML configuration."""
+        files = [ff for ff in _DATA_PATH.glob("*.yaml") if ff.name not in ignore_files]
+        files.sort()
+        return [cls(ff, ignore_funcs.get(ff.name, []))._parse_yaml() for ff in files]
+
+    def _parse_yaml(self) -> HeaderFile:
+        def read_docstring():
+            doc = []
+            with self._path.open("r", encoding="utf-8") as fid:
+                while True:
+                    line = fid.readline()
+                    if line.startswith("#"):
+                        doc.append(line.lstrip("#").strip())
+                    else:
+                        break
+            if doc and doc[0].startswith("This file is part of "):
+                return []
+            return doc
+
+        config = read_config(self._path)
+        recipes = []
+        prefix = config["prefix"]
+        base = config["base"]
+        parents = config.get("parents", [])
+        derived = config.get("derived", [])
+        for recipe in config["recipes"]:
+            if recipe['name'] in self._ignore_funcs:
```

Comment:
```suggestion
            if recipe["name"] in self._ignore_funcs:
```
I'd suggest running Black/Ruff on these changes if they don't cause too many other spurious changes :smile:

Reply:
While I believe we dropped the preference for `"` in Python strings, I manually searched for `'` and replaced what I found.